### PR TITLE
Development

### PR DESCRIPTION
**Pull request by rvanheest**
@janvanmansum
A first PR for the Easy Split-Multi-Deposit application. This contains the basic stuff like parsing the input CSV and running the Springfield actions (copy files and creating XML).

At this point to program runs, but fails because the `CreateMetadataAction` is not yet implemented. To ignore this, remove `multideposit/Main.79`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <name>EASY Split Multi-Deposit</name>
     <description>Splits a Multi-Deposit into several deposit directories for subsequent ingest into the archive</description>
     <properties>
-        <main-class>nl.knaw.dans.easy.ps.Main</main-class>
+        <main-class>nl.knaw.dans.easy.multiDeposit.Main</main-class>
```
Comment (janvanmansum) [inline pom.xml:33]: Please, use all-lowercase package names.

File: README.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ Steps:
         mvn install
 
 
-[Multi-Deposit Instructions]: multi-deposit-instructions.html
+[Multi-Deposit Instructions]: multi-deposit-instructions.md
```
Comment (janvanmansum) [inline README.md:77]: Don't forget to merge my meta-pull-request that updates the specs.

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    log.debug("Using the following settings: {}", settings)
+
+    settings
+  }
+}
+
+class ScallopCommandLine(conf: Config, args: Array[String]) extends ScallopConf(args) {
+  import nl.knaw.dans.easy.multiDeposit.{CommandLineOptions => cmd}
+
+  printedName = "process-multi-deposit"
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala:53]: `easy-split-multi-deposit`

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  version(s"$printedName ${Version()}")
+  banner("""Utility to process a Submission Information Package prior to ingestion into the DANS EASY Archive
+           |
+           |Usage: process-sip.sh [{--output-deposits-dir|-d} <dir>][{--springfield-inbox|-s} <dir>] <multi-deposit-dir>
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala:57]: `process-sip.sh` -> `$printedName`

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+           |Options:
+           |""".stripMargin)
+
+  lazy val multiDepositDir = trailArg[File](name = "multi-deposit-dir", required = true,
+    descr = "Directory containing the Submission Information Package to process. "
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala:62]: Submission Information Package -> Multi-Deposit

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+  def parse(args: Array[String]): Settings = {
+    log.debug("Loading application.conf ...")
+    val conf = ConfigFactory.load
+    log.debug("Parsing command line ...")
+    val opts = new ScallopCommandLine(conf, args)
+
+    val settings = Settings(
+      appHomeDir = new File(Option(System.getProperty("app.home"))
+        .getOrElse(Properties.propOrNull("process.sip.home"))),
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/CommandLineOptions.scala:39]: Remove `.getOrElse` part.

File: README.md (hunk 2 above)
Comment (rvanheest) [inline README.md:77]: Will do, please have a final look at this one: https://github.com/rvanheest/easy-split-multi-deposit/pull/1/files#r53905378

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/actions/CreateSpringfieldAction.scala [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import nl.knaw.dans.easy.multiDeposit.{Dataset, _}
+import nl.knaw.dans.easy.multiDeposit.actions.CreateSpringfieldAction._
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.PrettyPrinter
+
+case class Video(name: String, fileSip: Option[String], subtitles: Option[String])
+
+case class CreateSpringfieldAction(row: Int, datasets: Datasets)(implicit settings: Settings) extends Action(row) {
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/actions/CreateSpringfieldAction.scala:29]: Should be plural: `CreateSpringfieldActions`. The "Actions" here is the last part of the generated xml file `springfield-actions.xml` and does not refer to the fact that this class extends `Action`. I notice that the new `Action` subclasses you created have names ending in `Action`. Probably best remove that suffix. From the names being verb-object pairs it should be obvious that they encapsulate actions.

File: src/main/scala/nl/knaw/dans/easy/multiDeposit/Main.scala [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+  implicit class DatasetsExtensionMethods(val datasets: Observable[Datasets]) extends AnyVal {
+    def getActions(implicit s: Settings) = datasets.flatMap(Main.getActions)
+  }
+
+  def getActions(dss: Datasets)(implicit s: Settings): Observable[Action] = {
+    log.info("Compiling list of actions to perform ...")
+    val tryActions = dss.flatMap(getDatasetActions).toList :+
+      Success(CreateSpringfieldAction(-1, dss)) // SpringfieldAction runs on multiple rows, so -1 here
+
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/multiDeposit/Main.scala:67]: I am normally all for it to have as few comments as possible, but I think the next part needs some clarification.

**Pull request merged**